t/issue/948.t

```diff
@@ -0,0 +1,81 @@
+use strict;
+use warnings;
+
+use Rex::Virtualization;
+use Test::More tests => 6;
+use Data::Dumper;
+use Rex::Helper::Run;
+use Rex::Commands::File;
+use Rex::Commands::Fs;
+use Rex::Commands::Run;
+
+my $image_format = "raw";
+
+sub get_image_format { return $image_format; }
+
+my $count_file = 0;
+my $count_exec = 0;
+
+no warnings 'redefine';
+
+sub Rex::Commands::File::file {
```

Comment:
I'm not sure if I feel fully comfortable with the subroutine redefines in the tests. Though I agree that sooner or later we would need some mocking in the test suite, and this can be a precursor to it.

Can you maybe add a `#TODO: implement mocking and/or test this without subroutine redefining`, or similar, so at least it's marked as a task for the future?